src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -120,43 +120,59 @@ def validate_import_feature(feature):
 
 # File <-> List of KeyValue object
 
-
-def __read_kv_from_file(file_path,
-                        format_,
-                        separator=None,
-                        prefix_to_add="",
-                        depth=None,
-                        content_type=None):
+def __read_with_appropriate_encoding(file_path, format_):
     config_data = {}
     try:
+        with io.open(file_path, 'r', encoding='utf-8') as config_file:
+            if format_ == 'json':
+                config_data = json.load(config_file)
+
+            elif format_ == 'yaml':
+                for yaml_data in list(yaml.safe_load_all(config_file)):
+                    config_data.update(yaml_data)
+
+            elif format_ == 'properties':
+                config_data = javaproperties.load(config_file)
+                logger.debug("Importing feature flags from a properties file is not supported. If properties file contains feature flags, they will be imported as regular key-values.")
+
+    except ValueError:
+        # Try detecting the encoding instead of using the default utf-8 encoding
```

Comment:
We can put try except logic in __read_kv_from_file. This method can just take an encoding and read the file.